activerecord/test/cases/strict_loading_test.rb

```diff
@@ -53,6 +54,10 @@ def test_strict_loading_n_plus_one_only_mode
     developer.strict_loading!(mode: :n_plus_one_only)
     assert_predicate developer, :strict_loading?
 
+    # Does not eager load a child association
+    developer.projects.first
+    assert_not_predicate developer.projects, :loaded?
+
```

Comment:
There already is a lot going on in this test.
Moving the described issue to a new test case would make the problem more clear.

Reply:
Thanks for taking a look, @p8. I broke this into a separate test case in 4c90fdf1ce1634cdd3a56214dc7ec96cf2501a3e